File: leetcode/python/1.py
## Leetcode 1 - Two Sum
## Nov. 24, 2023

from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ## one pass hashmap
        visited = {}
        for i in range(len(nums)):
            c = target - nums[i]
            if c in visited:
                return [i, visited[c]]
            else:
                visited[nums[i]] = i

File: leetcode/c++/1.cpp
// Leetcode 1 - Two Sum
// Nov. 24, 2023

#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> visited;
        for (int i = 0; i < nums.size(); i++) {
            int c = target - nums[i];
            if (visited.find(c) == visited.end()) {
                visited[nums[i]] = i;
            } else {
                return {visited[c], i};
            }
        }
        return {-1,-1};
    }
};